aea/configurations/base.py

```diff
@@ -228,8 +228,7 @@ class PackageConfiguration(Configuration, ABC):
     package_type: PackageType
 
     FIELDS_ALLOWED_TO_UPDATE: FrozenSet[str] = frozenset(["build_directory"])
-    FIELDS_WITH_NESTED_FIELDS: FrozenSet[str] = frozenset()
-    NESTED_FIELDS_ALLOWED_TO_UPDATE: FrozenSet[str] = frozenset()
```

Comment:
We don't need the `NESTED_FIELDS_ALLOWED_TO_UPDATE` anymore because this makes performing overrides less flexible. `FIELDS_WITH_NESTED_FIELDS` -> `OVERRIDABLE_FIELDS` is just a rename but it seems more suitable.

Reply:
I was a bit confused on the usage of `FIELDS_ALLOWED_TO_UPDATE` so I kept it and introduced the `OVERRIDABLE_FIELDS`, But it has the same purpose as the `OVERRIDABLE_FIELDS`. So I reverted the changes and now we are using `FIELDS_ALLOWED_TO_UPDATE` to generate the overridables metadata.